src/securityPolicy.c

```diff
@@ -362,14 +365,18 @@ security_policy_t policyForSignTxWitness(
 {
 	DENY_UNLESS(is_valid_witness(pathSpec));
 
+	WARN_UNLESS(has_reasonable_account(pathSpec));
```

Comment:
won't this technically make it possible (with a warning though) to sign a stake pool registration with a spending key for an "unreasonable" account number?

Reply:
how I imagine the logic:

```
DENY_UNLESS(is_valid_witness(pathSpec));

if (isSigningPoolRegistrationAsOwner) {
  DENY_UNLESS(is_valid_stake_pool_owner_path(pathSpec));
}
DENY_IF(is_too_deep(pathSpec));
WARN_UNLESS(has_reasonable_account_and_address(pathSpec));
```

i.e. I think the deny policies should come first, otherwise they can be invalidated by the WARN_IF policies if they happen to overlap (and the deny action should probably have precedence)